lib/ramble/ramble/experiment_set.py

```diff
@@ -532,18 +559,46 @@ def _ingest_experiments(self):
             app_inst.set_variables(experiment_vars, self)
             app_inst.set_env_variable_sets(ordered_env_variables)
             app_inst.set_internals(merged_internals)
-            app_inst.add_expand_vars(self._workspace)
-
+            app_inst.set_template(is_template)
+            app_inst.set_chained_experiments(merged_chained_experiments)
             self.experiments[experiment_namespace] = app_inst
+            self.experiment_order.append(experiment_namespace)
+
+    def finalize_set(self):
```

Comment:
It doesn't really render though, it only renders experiments defined in a chain (and honestly, it's only done here because you can chain in experiments that haven't been rendered when this experiment is rendered).

Another option would be to make this an internal thing, and call it when other utility methods are called (like `all_experiments`). As in:

```
def all_experiments(self):
  if not self._is_finalized:
    self._finalize_set()
  for ...
    yield exp, inst
```

Thoughts?

Reply:
Changes look good!